### ISSUE TITLE
keyboard access for resizing and reordering columns for DetailsList

### ISSUE THREAD

**Issue by smhigley** [microsoft/fluentui#18788]
## Problem:

DetailsList provides drag to resize and reorder columns, but doesn't provide a way to perform those actions with the keyboard, or with a pointer without drag/drop (so failing [pointer gestures](https://www.w3.org/WAI/WCAG21/Understanding/pointer-gestures.html) and [keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)).

There aren't currently any well-established patterns for keyboard control of resizable widgets (e.g. textarea doesn't seem to have any keyboard resizing ability), so we're starting more or less from scratch. Excel does it through menus and a settings dialog, which can't translate directly to a standalone widget.

## Proposed solution:

After a discussion with @scottaohara, I think the best approach would be the following:

1. Add a custom context menu to column headers with the following menuitems (adding sorting as well, because if we're adding a context menu, we may as well expose sorting in multiple ways):
    - Sort ascending
    - Sort descending
    - Resize column
    - Reorder column
2. Add a dialog for resizing or reordering the column, where a user can enter a new order index or pixel width.
3. Figure out how we want to handle layout and reflow when users have entered in fixed pixel widths for specific columns, and whether this differs from how we currently handle drag-to-resize.
4. Add a ctrl + arrow keys keyboard shortcut to resize the focused column header.
4. Potentially add a button to open the context menu that is only present when the column header has keyboard focus. Only-on-focus prevents it from being part of the column header's name at other times, and it would then be possible to access via touch.

### Potential problems:
- we should check how much adding ContextualMenu and Dialog will affect the bundle size
- we should investigate what sorts of limits we should put on manually entering a column width (e.g. should there be a max?)

**Comment by ecraig12345** [comment]
For what it's worth, I don't think this ought to be baked in (unless there's a particularly compelling reason), but we could definitely add an example with a column header menu that does all this stuff.

**Comment by Sanankh** [comment]
@smhigley @ecraig12345  any timelines on fixing this?

**Comment by jeffersoneagley** [comment]
The fact that this keeps getting flagged in accessibility passes would seem to indicate this _should_ be part of the component itself. The component ought to be generally accessible out-of-the-box.

**Comment by jgage176** [comment]
I completely agree with @jeffersoneagley - Why would it only be an example that is added if it is required for EVERY single instance? Every instance they are saying it MUST support keyboard.

I get that there may need to be some sort of flag for enabling or disabling it or even a component that sits on top, like the `ShimmerDetailsList` - maybe there should be another one for this one? We don't want to break those that have already built this functionality. It should be a drop-in replacement (if possible) so people can just change their instances if they weren't doing extra and get the keyboard accessible version.

**Comment by smhigley** [comment]
@jeffersoneagley / @jgage176 the problem is that some if not most usage of DetailsList does make use of the resize/reorder functionality. If we built in the ContextualMenu, then all DetailsList users, including those not using resize/reorder, would still get ContextualMenu packaged with DetailsList anyway, which is a pretty hefty bundle size increase. It essentially comes down to the kitchen-sink approach of putting all functionality in the base DetailsList implementation. We're moving away from that for v9, but it means it's very hard to add functionality to the v8 DetailsList.
